### PR TITLE
fix(compiler): generated CSS files suffixed with ngstyle.

### PR DESCRIPTION
**Pull request by hansl**
**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [X] Feature (see #13141)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [X] Maybe ?  I'm not sure.
- [ ] No

### CODE REVIEW

File: modules/@angular/compiler-cli/src/codegen.ts [hunk 1]
```diff
@@ -21,9 +21,9 @@ import {CompilerHost, CompilerHostContext, ModuleResolutionHostAdapter} from './
 import {PathMappedCompilerHost} from './path_mapped_compiler_host';
 import {Console} from './private_import_core';
 
-const GENERATED_FILES = /\.ngfactory\.ts$|\.css\.ts$|\.css\.shim\.ts$/;
+const GENERATED_FILES = /\.ngfactory\.ts$|\.ngstyle\.ts$/;
```
Comment (tbosch) [inline modules/@angular/compiler-cli/src/codegen.ts]: You need 2 suffixes, one for shimmed and one for non shimmed.
e.g. `.ngstyle.ts` and `.ngstyle.shim.ts`

Comment (hansl) [inline modules/@angular/compiler-cli/src/codegen.ts]: I'll use `.shim.ngstyle.ts` instead, so this regex stays simplified.

**Comment by hansl** [comment]
Seems to work fine, need to pass on google3.

**Comment by hansl** [comment]
@vicb here's the test run for this: https://test.corp.google.com/ui#id=OCL:141921829:BASE:141925779:1481661350322:e67982e8

Note the CL contains a change that's not part of this PR. Check with me for more details.

**Comment by hansl** [comment]
@tbosch PTAL. I made this pass with google3.

File: modules/@angular/compiler/src/jit/compiler.ts [hunk 2]
```diff
@@ -342,7 +342,7 @@ export class JitCompiler implements Compiler {
     if (!this._compilerConfig.useJit) {
       return interpretStatements(result.statements, result.stylesVar);
     } else {
-      return jitStatements(`/${result.meta.moduleUrl}.css.js`, result.statements, result.stylesVar);
+      return jitStatements(`/${result.meta.moduleUrl}.css.ngstyle.js`, result.statements, result.stylesVar);
```
Comment (tbosch) [inline modules/@angular/compiler/src/jit/compiler.ts]: Maybe just use `.ngstyle.js`, drop `.css`. This filename is shown to the user when they open dev tools, having too many suffixes is probably more confusing than helping.

**Comment by tbosch** [comment]
Please change the one string above and squash, otherwise looks good.

File: modules/@angular/compiler/src/jit/compiler.ts (hunk 2 above)
Comment (hansl) [inline modules/@angular/compiler/src/jit/compiler.ts]: Done and repushed.

**Comment by hansl** [comment]
@vicb Ping me when this is cherry picked for google3; you need to make a slight change as per the CL above.

**Pull request merged**